### PR TITLE
IA-3908: Support `deleteDisk` param when deleting runtime via v2 routes

### PR DESCRIPTION
**Pull request by lucymcnatt**
Behavior:
- with no argument and deleteDisk=false the PD isn't deleted
- when deleteDisk=true it is deleted

Implementation:
**Front Leo**:
- accept query param
- Pass Boolean properly in pubsub message to back Leo

**Back Leo**:
- Azure pubsub handler class: handles deleting runtime
- Make delete disk call conditional

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoApiClient.scala
Comment (Qi77Qi) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoApiClient.scala]: let's not provide default here....just so the caller can make an intentional decision whether the disk should be deleted

File: http/src/main/resources/swagger/api-docs.yaml
Comment (Qi77Qi) [inline http/src/main/resources/swagger/api-docs.yaml]: maybe `Whether or not disk should be deleted upon runtime deletion`?

**Comment by lucymcnatt** [comment]
jenkins retest

**Comment by lucymcnatt** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoApiClient.scala
Comment (lucymcnatt) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoApiClient.scala]: sounds good, I'll change it for `deleteRuntimeV2WithWait` also

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala
Comment (Qi77Qi) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala]: we can remove the network call now since we no longer create network resources for each VM creation

File: .vscode/settings.json
Comment (LizBaldo) [inline .vscode/settings.json]: Should this be pushed here? i usually have `.vscode` stuff in gitignore

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala
Comment (LizBaldo) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala:564]: Why was this deleted? Should we keep handling network deletion?

File: .vscode/settings.json
Comment (lucymcnatt) [reply .vscode/settings.json]: oh oops I meant to put that in my gitignore, I must've accidently pushed this up in one of my recent commits

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala
Comment (lucymcnatt) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala:564]: this was in response to Qi's comment here: https://github.com/DataBiosphere/leonardo/pull/3095#discussion_r1067357873

Comment (LizBaldo) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala:564]: Gotcha, sorry for missing it 😵‍💫

**Review by LizBaldo — APPROVED** [review]
Thanks for addressing my comments! LGTM, if you can replicate it locally via swagger you have my 👍

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala
Comment (Qi77Qi) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala:545]: nit, can we move this to after `logger.info(ctx.loggingCtx)("runtime is deleted successfully")` line

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandlerSpec.scala
Comment (Qi77Qi) [inline http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandlerSpec.scala]: this is not needed

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala
Comment (lucymcnatt) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala:545]: yeah np, easy enough!

**Review by Qi77Qi — APPROVED** [review]
gonna give a thumb and will trust @lucymcnatt has tested locally before merging or test it after merging on dev if you're super confident :)

**Comment by lucymcnatt** [comment]
tested locally and it behaves as expected! 
-  with no argument and deleteDisk=false the PD isn't deleted
-  when deleteDisk=true it is deleted

**Comment by lucymcnatt** [comment]
> 

Yes I tested locally (and will double check JIC), and on dev after merging!

**Comment by lucymcnatt** [comment]
> 

I'll double check locally JIC, and will on dev after merging!

**Comment by lucymcnatt** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
```
org.broadinstitute.dsde.workbench.leonardo.rstudio.RStudioSpec.RStudioSpec test Rmd file syncing
```

jenkins retest

**Comment by lucymcnatt** [comment]
jenkins retest

**Comment by lucymcnatt** [comment]
jenkins retest

**Comment by lucymcnatt** [comment]
jenkins retest

**Comment by lucymcnatt** [comment]
jenkins retest

**Pull request merged**
